### ISSUE TITLE
MD029 false positive with display math

### ISSUE THREAD

**Issue by palapapa** [DavidAnson/vscode-markdownlint#333]
The following:
```markdown
# 1

1. a

    1. asd

    $$
    123
    $$

    2. asd
```
triggers MD029, whereas the following:
```markdown
# 1

1. a

    1. asd
    $$
    123
    $$

    2. asd
```
doesn't.

**Comment by DavidAnson** [comment]
The math content is not properly indented to be part of the nested list. In the second example, it immediately follows the list item content and so is treated as part of it. In the first example, the blank line means the math content ends the nested list. You can experiment with both forms in Source view here to see what I mean: https://markdown-it.github.io/

I believe the current behavior of markdownlint is correct based on how Markdown treats these two situations differently.

**Comment by palapapa** [comment]
Sorry, it was a mistake. The correct way to write it is:
```markdown
# 1

1. a

    1. asd

        $$
        123
        $$

    2. asd
```
